src/components/input/ButtonBase.js

```diff
@@ -56,8 +58,19 @@ const ButtonBaseNext = function ButtonBase({
     <button
       role={role ?? 'button'}
       {...ariaProps}
+      data-base-component="ButtonBase"
+      /* data-component will be overwritten unless this component is used directly */
+      data-component="ButtonBase"
       {...htmlAttributes}
-      className={className}
+      className={classNames(
+        {
+          'focus-visible-ring rounded-sm': !unstyled,
+          'transition-colors': !unstyled,
+          // Set layout for button content
+          'whitespace-nowrap flex items-center': !unstyled,
```

Comment:
So, these are the canonized "core styles" for a button in our world. But we don't apply them if `unstyled` is set...